File: src/BusinessLogic/Handlers/MessageHandlers/StartMessageHandler.cs
using Bounan.Bot.TelegramBot.Telegram;
using Microsoft.Extensions.Logging;
using Telegram.Bot;
using Telegram.Bot.Types;
using Telegram.Bot.Types.ReplyMarkups;

namespace Bounan.Bot.BusinessLogic.Handlers.MessageHandlers;

public class StartMessageHandler(
    ILogger<StartMessageHandler> logger,
    ITelegramBotClient botClient) : IMessageHandler
{
    public static bool CanHandle(Message message) => message.Text?.StartsWith("/start") ?? false;

    public Task HandleAsync(Message message, CancellationToken cancellationToken)
    {
        ArgumentNullException.ThrowIfNull(message);
        ArgumentNullException.ThrowIfNull(message.Chat);
        ArgumentNullException.ThrowIfNull(message.Chat.Id);
        ArgumentNullException.ThrowIfNull(message.Text);

        logger.LogInformation("Handling start message {Text}", message.Text);

        return botClient.SendTextMessageAsync(
            message.Chat.Id,
            "Напиши название аниме или нажми на кнопку \"Искать\", и я найду его для тебя",
            replyMarkup: new InlineKeyboardMarkup([
                new InlineKeyboardButton("Искать") { SwitchInlineQueryCurrentChat = string.Empty }
            ]),
            cancellationToken: cancellationToken);
    }
}

File: src/handlers/on-webhook/event-handlers/message-handlers/start-message-handler.ts
import { sendMessage, Message } from '@lightweight-clients/telegram-bot-api-lightweight-client';
import { MessageHandler } from '../query-handler';
import { assert } from '../../../../shared/helpers/assert';
import { Texts } from '../../../../shared/telegram/texts';

const canHandle = (message: Message): boolean => message.text?.startsWith('/start') || false;

const handler: MessageHandler = async (message) => {
    assert(!!message?.chat?.id);
    assert(!!message?.text);

    console.log('Handling start message', message.text);

    const result = await sendMessage({
        chat_id: message.chat.id,
        text: Texts.Start,
        reply_markup: {
            inline_keyboard: [
                [
                    { text: Texts.Button__Search, switch_inline_query_current_chat: '' },
                ],
            ],
        },
    });
    assert(result.ok, () => JSON.stringify(result));
};

export const startMessageHandler = {
    canHandle,
    handler,
};